pulsar-client/src/main/java/com/yahoo/pulsar/client/impl/ConsumerBase.java

```diff
@@ -72,9 +72,14 @@ protected ConsumerBase(PulsarClientImpl client, String topic, String subscriptio
         }
         this.listenerExecutor = listenerExecutor;
         this.pendingReceives = Queues.newConcurrentLinkedQueue();
-        if (conf.getAckTimeoutMillis() != 0) {
+
+        // initialize unAckedMessageTracker if stats (client-ack-stats uses it to track ack-message for cumulative
+        // ack) or AckTimeOut-Task enabled 
+        if (client.getConfiguration().getStatsIntervalSeconds() > 0 || conf.getAckTimeoutMillis() != 0) {
```

Comment:
Made changes with first option where we will not initialize ```unAckedMessageTracker``` for ack-stats counter.